[Reporting/Docs] Add clarity to reporting concepts (#125175)
elastic/kibana: docs/settings/reporting-settings.asciidoc

* <<general-reporting-settings,Enable the {report-features}>>
* <<encryption-keys,Configure the encryption key>>
* <<reporting-job-queue-settings,Manage background jobs>>
* <<reporting-capture-settings,Capture screenshots>>
* <<reporting-network-policy,Restrict requests with a Reporting network policy>>
* <<reporting-csv-settings,Increase the byte limit for CSV exports>>
* <<reporting-kibana-server-settings,Control how the {report-features} communicate with the {kib} server>>

[float]
[[general-reporting-settings]]
==== Enable reporting

[[xpack-enable-reporting]]`xpack.reporting.enabled` {ess-icon}::

> When `true` (default), enables the {report-features}. Set this to `false` to disable {report-features} entirely.

Reviewer: ```suggestion
When `true`, enables the {report-features}. Set this to `false` to disable {report-features} entirely. The default is `true`.
```

`xpack.reporting.queue.indexInterval`::
How often the index that stores reporting jobs rolls over to a new index. Valid values are `year`, `month`, `week`, `day`, and `hour`. Defaults to `week`.

[[xpack-reportingQueue-pollEnabled]] `xpack.reporting.queue.pollEnabled` {ess-icon}::

> Set to `true` (default) to enable the {kib} instance to poll Elasticsearch for pending jobs and claim them for

Reviewer: ```suggestion
When `true`, enables the {kib} instance to poll {es} for pending jobs and claim them for
```

> execution. Setting this to `false` allows the {kib} instance to only add new jobs to the reporting queue, list

Reviewer: ```suggestion
execution. When `false`, allows the {kib} instance to only add new jobs to the reporting queue, list
```

jobs, and provide the downloads to completed reports through the UI. This requires a deployment where at least

> one other Kibana instance in the Elastic cluster has this setting to `true`. The default is `true`.

Author: ```suggestion
one other {kib} instance in the Elastic cluster has this setting to `true`. The default is `true`.
```

> one other Kibana instance in the Elastic cluster has this setting to `true`.

Reviewer: ```suggestion
one other Kibana instance in the Elastic cluster has this setting to `true`. The default is `true`.
```

> one other Kibana instance in the Elastic cluster has this setting to `true`.

Reviewer: ```suggestion
one other {kib} instance in the Elastic cluster has this setting set to `true`.
```

[NOTE]
============================================================================

> The `xpack.reporting.roles` settings are for a deprecated system of access control in Reporting. Turning off this feature allows API Keys to generate reports, and allows reporting access through Kibana application privileges. We recommend you explicitly turn off reporting's deprecated access control feature by adding `xpack.reporting.roles.enabled: false` in kibana.yml. This will enable you to create custom roles that provide application privileges for reporting, as described in <<grant-user-access, granting users access to reporting>>.

Reviewer: ```suggestion
The `xpack.reporting.roles` settings are for a deprecated system of access control in Reporting. Turning off this feature allows API Keys to generate reports, and allows reporting access through {kib} application privileges. We recommend you explicitly turn off reporting's deprecated access control feature by adding `xpack.reporting.roles.enabled: false` in kibana.yml. This will enable you to create custom roles that provide application privileges for reporting, as described in <<grant-user-access, granting users access to reporting>>.
```

============================================================================

[[xpack-reporting-roles-enabled]] `xpack.reporting.roles.enabled`::
deprecated:[7.14.0,The default for this setting will be `false` in an upcoming version of {kib}.] Sets access control to a set of assigned reporting roles, specified by `xpack.reporting.roles.allow`. Defaults to `true`.

`xpack.reporting.roles.allow`::
deprecated:[7.14.0] In addition to superusers, specifies the roles that can generate reports using the {ref}/security-api.html#security-role-apis[{es} role management APIs]. Requires `xpack.reporting.roles.enabled` to be `true`. Defaults to `[ "reporting_user" ]`.

[float]
[[reporting-kibana-server-settings]]
==== {kib} server settings

To generate screenshots for PNG and PDF reports, Reporting opens the {kib} web interface using a local
connection on the server. In most cases, using a local connection to the {kib} server presents no issue. If
you prefer the headless browser to connect to {kib} using a specific hostname, there are a number of

> settings which can allow the headless browser to connect to {kib} through a proxy, rather than directly.

Reviewer: ```suggestion
settings that allow the headless browser to connect to {kib} through a proxy, rather than directly.
```

[NOTE]
============
The `xpack.reporting.kibanaServer` settings are optional. Take caution when editing these settings. Adding

> these settings brings a new category of reasons that {report-features} might not work. The clues to the reason

Reviewer: ```suggestion
these settings can cause the {report-features} to fail. If report fail,
```

> for failing reports will require inspecting the server logs: the full {kib} URL that Reporting is attempting to

Reviewer: ```suggestion
inspect the server logs. The full {kib} URL that Reporting is attempting to
```
